pkg/system/phase2_creating.go

```diff
@@ -61,6 +62,15 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 		return nil
 	}
 
+	if r.NooBaa.Spec.DBType == "postgres" {
+		r.NooBaaDB = util.KubeObject(bundle.File_deploy_internal_statefulset_postgres_db_yaml).(*appsv1.StatefulSet)
+		r.ServiceDb = util.KubeObject(bundle.File_deploy_internal_service_postgres_db_yaml).(*corev1.Service)
+		r.NooBaaDB.Namespace = r.Request.Namespace
+		r.NooBaaDB.Name = r.Request.Name + "-db"
+		r.ServiceDb.Namespace = r.Request.Namespace
+		r.ServiceDb.Name = r.Request.Name + "-db"
+	}
+
```

Comment:
Why not create both DB statefulsets and DB services on the reconciler (same name and namespace) and choose which one to update to k8s ?

Reply:
You are also not loading the current resource from k8s which mean that any allowed changes to the sts or service will be overwritten